src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedArray = '';
+
+    for (const item of this) {
+      if (item === null || item === undefined) {
+        joinedArray += '' + separator;
+      } else {
+        if (item !== this[this.length - 1]) {
+        }
+        joinedArray += item + '' + separator;
+      }
+    }
+
+    if (joinedArray[joinedArray.length - 1] === separator) {
+      joinedArray = joinedArray.slice(0, -1);
+    }
+
+    if (separator === null) {
+      joinedArray = joinedArray.slice(0, -4);
+    }
+
+    if (typeof separator === 'object' && separator !== null) {
+      joinedArray = joinedArray.slice(0, -15);
+    }
```

Comment:
This check works, but it is too complicated.
Also, it won't work, if separator will be a string with `length > 1`

How about we change logic a little bit and just won't add separator after the last array element?